### ISSUE TITLE
Chaiscript may not be a good choice

### ISSUE THREAD

**Issue by ghost** [Ghabry/EasyScript#2]
Hello

All implementations of chaiscript are written in header files, which causes extremely slow compilation speeds for every cpp file that includes chaiscript. One possible solution is to find a way to optimize it so that each cpp file that includes chaiscript does not need to recompile the chaiscript portion repeatedly (using ccache also does not solve this issue).

**Comment by Ghabry** [comment]
yeah, I'm also extremely disappointed by the compile times of chaiscript.

The C++ integration is great but everything else is lacking. Besides slow compile times the runtime speed is also very slow (though this doesn't really matter for such a tool but makes it imo unusable as an embedded scripting language)
